build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetPreview.cs

```diff
@@ -65,68 +62,188 @@ protected override async Task<bool> Execute (Context context)
 			if (!success) {
-				if (status == HttpStatusCode.NotFound)
-					Log.ErrorLine ("dotnet-install URL not found");
-				else
-					Log.ErrorLine ("Failed to obtain dotnet-install size. HTTP status code: {status} ({(int)status})");
+				if (status == HttpStatusCode.NotFound) {
+					Log.ErrorLine ($"dotnet archive URL {archiveUrl} not found");
+				} else {
+					Log.ErrorLine ($"Failed to obtain dotnet archive size. HTTP status code: {status} ({(int)status})");
+				}
+
 				return false;
 			}
 
+			string tempArchiveDestinationPath = archiveDestinationPath + "-tmp";
+			Utilities.DeleteFile (tempArchiveDestinationPath);
+
 			DownloadStatus downloadStatus = Utilities.SetupDownloadStatus (context, size, context.InteractiveSession);
-			Log.StatusLine ($"  {context.Characters.Link} {dotnetScriptUrl}", ConsoleColor.White);
-			await Download (context, dotnetScriptUrl, dotnetScriptPath, "dotnet-install", Path.GetFileName (dotnetScriptUrl.LocalPath), downloadStatus);
+			Log.StatusLine ($"  {context.Characters.Link} {archiveUrl}", ConsoleColor.White);
+			await Download (context, archiveUrl, tempArchiveDestinationPath, "dotnet archive", Path.GetFileName (archiveUrl.LocalPath), downloadStatus);
 
-			if (!File.Exists (dotnetScriptPath)) {
-				Log.ErrorLine ($"Download of dotnet-install from {dotnetScriptUrl} failed");
+			if (!File.Exists (tempArchiveDestinationPath)) {
 				return false;
 			}
 
-			var type = runtimeOnly ? "runtime" : "SDK";
-			Log.StatusLine ($"Installing dotnet {type} '{version}'...");
+			Utilities.CopyFile (tempArchiveDestinationPath, archiveDestinationPath);
+			Utilities.DeleteFile (tempArchiveDestinationPath);
+
+			return true;
+		}
 
+		string[] GetInstallationScriptArgs (string version, string dotnetPath, string dotnetScriptPath, bool onlyGetUrls, bool runtimeOnly)
+		{
+			List<string> args;
 			if (Context.IsWindows) {
-				var args = new List<string> {
+				args = new List<string> {
 					"-NoProfile", "-ExecutionPolicy", "unrestricted", "-file", dotnetScriptPath,
 					"-Version", version, "-InstallDir", dotnetPath, "-Verbose"
 				};
-				if (runtimeOnly)
+				if (runtimeOnly) {
 					args.AddRange (new string [] { "-Runtime", "dotnet" });
+				}
+				if (onlyGetUrls) {
+					args.Add ("-DryRun");
+				}
 
-				return Utilities.RunCommand ("powershell.exe", args.ToArray ());
-			} else {
-				var args = new List<string> {
-					dotnetScriptPath, "--version", version, "--install-dir", dotnetPath, "--verbose"
-				};
-				if (runtimeOnly)
-					args.AddRange (new string [] { "-Runtime", "dotnet" });
+				return args.ToArray ();
+			}
+
+			args = new List<string> {
+				dotnetScriptPath, "--version", version, "--install-dir", dotnetPath, "--verbose", "--dry-run"
+			};
+
+			if (runtimeOnly) {
+				args.AddRange (new string [] { "-Runtime", "dotnet" });
+			}
+			if (onlyGetUrls) {
+				args.Add ("--dry-run");
```

Comment:
don't we already add `--dry-run` on line 163? Do we need this as well or do we need the arg on line 163?

Reply:
Good eye, it's a mistake, will fix